src/D2L.Security.OAuth2/Keys/Caching/InMemoryPublicKeyCache.net50.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using D2L.Security.OAuth2.Keys.Default;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace D2L.Security.OAuth2.Keys.Caching {
+	internal sealed partial class InMemoryPublicKeyCache : IInMemoryPublicKeyCache {
+		private readonly IMemoryCache m_cache;
+
+		public InMemoryPublicKeyCache() : this(
+			new MemoryCache( new MemoryCacheOptions() ) ) {}

```

Comment:
This is a difference in behaviour: in the old impl we use the default memory cache which is a global variable.... we could replicate that (mostly) with a `static Lazy<MemoryCache>` that we pull from here.... but I think this is fine...

Reply:
In the LMS we invoke `AccessTokenValidatorFactory` on each request which news up a new cache object

https://github.com/Brightspace/D2L.Security.OAuth2/blob/9fbe981668356eb39ae875b4da02bfdae30004fc/src/D2L.Security.OAuth2/Validation/AccessTokens/AccessTokenValidatorFactory.cs#L51

Believe we're thus relying on the global for caching to be effective.

Reply:
Ok

Reply:
Done